Examples/Detectors/DD4hepDetector/src/DD4hepDetector.cpp

```diff
@@ -25,18 +24,24 @@
 
 namespace ActsExamples {
 
-DD4hepDetector::DD4hepDetector(
-    Acts::GeometryContext geometryContext,
-    std::vector<std::shared_ptr<const Acts::DetectorElementBase>> detectorStore,
-    std::shared_ptr<const Acts::TrackingGeometry> gen1Geometry,
-    std::shared_ptr<Acts::Experimental::Detector> gen2Geometry,
-    std::vector<std::shared_ptr<IContextDecorator>> contextDecorators,
-    std::shared_ptr<dd4hep::Detector> detector)
-    : PreConstructedDetector(std::move(geometryContext),
-                             std::move(detectorStore), std::move(gen1Geometry),
-                             std::move(gen2Geometry),
-                             std::move(contextDecorators)),
-      m_detector(std::move(detector)) {}
+DD4hepDetector::DD4hepDetector(const Config& cfg)
+    : Detector(Acts::getDefaultLogger("DD4hepDetector", cfg.logLevel)),
+      m_cfg(cfg) {
+  if (m_cfg.xmlFileNames.empty()) {
+    throw std::invalid_argument("Missing DD4hep XML filenames");
+  }
+
+  m_nominalGeometryContext = Acts::GeometryContext();
```

Comment:
If the constructor now constructs the geometry, can we make this an argument that's passed in? We can still hang on to it in the class to that one can always refer back to the construction context.

Reply:
After discussion we agreed that this is not great but for now the easiest way to get a nominal context that is driven by the user and the detector. We will have to revisit this in the future when we do ODD mis-alignment.